### ISSUE TITLE
gui.safeAppExit is called too early making it impossible for add-ons to cleanup their own GUI

### ISSUE THREAD

**Issue by lukaszgo1** [nvaccess/nvda#12325]
Regression from #12286

### Steps to reproduce:
1. Install an add-on which adds menu items to NVDA menu such as Unicorn DVC, NVDA Remote etc.
2. Exit NVDA and inspect the log from the previous run.
### Actual behavior:
Entries similar to the following can be seen:
```
DEBUG - gui.safeAppExit (17:40:54.104) - MainThread (5908):
destroying system tray icon and menu
DEBUG - gui.safeAppExit (17:40:54.124) - MainThread (5908):
destroying main frame during exit process
DEBUG - gui.safeAppExit (17:40:54.124) - MainThread (5908):
closing window <wx._core.Frame object at 0x0B26C620> during exit process
DEBUG - gui.safeAppExit (17:40:54.124) - MainThread (5908):
closing window <gui.ExitDialog object at 0x0B54FE90> during exit process
INFO - core.main (17:40:54.144) - MainThread (5908):
Exiting
DEBUG - core._terminate (17:40:54.144) - MainThread (5908):
Terminating watchdog
DEBUG - core._terminate (17:40:54.144) - MainThread (5908):
Terminating global plugin handler
ERROR - globalPluginHandler.terminate (17:40:54.144) - MainThread (5908):
Error terminating global plugin <globalPlugins.unicorn.GlobalPlugin object at 0x09F20730>
Traceback (most recent call last):
  File "globalPluginHandler.py", line 41, in terminate
    plugin.terminate()
  File "userConfig\addons\unicornNVDA\globalPlugins\unicorn\__init__.py", line 132, in terminate
    gui.mainFrame.sysTrayIcon.menu.Remove(self.submenu_item)
RuntimeError: wrapped C/C++ object of type Menu has been deleted

```

### Expected behavior:

Add-ons should be able to clean up their own gui.

#### NVDA installed/portable/running from source:
From sources
#### NVDA version:
Current master as of 22--th of April
#### Windows version:
Windows 7 x64
#### Name and version of other software in use when reproducing the issue:
None
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Versions before #12286 do not have this `problem.`
#### If add-ons are disabled, is your problem still occurring?
No but this issue is about NVDA interaction with add-ons
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
N/A

**Comment by lukaszgo1** [comment]
@seanbudd  My first thought would be to make `gui.safeAppExit` responsible for terminating  our modules in the right order i.e. starting with `globalPluginHandler` `appModuleHandler` and only when they're terminated we should start terminating gui.

**Comment by seanbudd** [comment]
> @seanbudd My first thought would be to make `gui.safeAppExit` responsible for terminating our modules in the right order i.e. starting with `globalPluginHandler` `appModuleHandler` and only when they're terminated we should start terminating gui.

That would be my approach as well. I'm looking to revert #12286 and then continue the work there

**Comment by seanbudd** [comment]
Closed by #12326 and the fixed methodology implemented in #12342
